- Miner(event db)

ssardana96 opened the pull request:
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a branch at [0chain/system_test](https://github.com/0chain/system_test) then add and merge your test changes to master BEFORE merging this PR

ssardana08 committed:
- add miner(event db) first commit

ssardana08 committed:
- added Miner table to AutoMigrate
- new methods to add/overwrite miner
- getMinerNode updated to get miner node from Event DB
- added new stats to events

Sriep commented on code/go/0chain.net/smartcontract/minersc/miner.go:
```diff
@@ -133,23 +133,30 @@ func (msc *MinerSmartContract) AddMiner(t *transaction.Transaction,
-		}
-		update = true
-	}
+		//if err = updateMinersList(balances, allMiners); err != nil {
```
Why remove this block? We are only emitting events in smartcontracts, not changing how miners work.

ssardana96 replied:
done

Sriep commented on code/go/0chain.net/smartcontract/minersc/miner.go:
```diff
@@ -133,23 +133,30 @@ func (msc *MinerSmartContract) AddMiner(t *transaction.Transaction,
 		update = true
 	}
 
+	//if !msc.doesMinerExist(newMiner.GetKey(), balances) {
```
Again leave this block?

ssardana96 replied:
done

Sriep commented on code/go/0chain.net/smartcontract/minersc/models.go:
```diff
@@ -1218,13 +1221,34 @@ func (dmn *DKGMinerNodes) GetHashBytes() []byte {
 
 // getMinersList returns miners list
 func getMinersList(state cstate.StateContextI) (*MinerNodes, error) {
+
```
The only changes to the smart contract code should be to emit events. 

Smartcontracts should have no access to the eventDb object, this is the reason for https://github.com/0chain/0chain/issues/657, to prevent people trying to use methods like GetEventDb in smart contracts.

ssardana96 replied:
noted. So in this case is someone does GetEventDb() (even through not should be done), it should return nil, right?
That's the expected behavior?

Sriep commented on code/go/0chain.net/smartcontract/minersc/models.go:
```diff
@@ -821,6 +821,9 @@ type SimpleNode struct {
 
 	// LastHealthCheck used to check for active node
 	LastHealthCheck common.Timestamp `json:"last_health_check"`
+
```
I don't think we need this. Instead emit en event in [createMagicBlock](https://github.com/0chain/0chain/blob/1e8779f639d24bcd832baaf07adbd2e31915b066/code/go/0chain.net/smartcontract/minersc/dkg.go#L798) with the `Status` update converting to a boolean Active.

Sriep reviewed (changes requested):
Expect the only change made to smartcontracts is to emit events. We should not be changing how smartcontracts work.

ssardana08 committed:
- removed Active from SimpleNode
- changed TotalStaked in edb Table from int64 to Balance
- no logic change in smart contracts, just emitting events

ssardana08 committed:
- emitUpdateMiner added
- for now removing Active. Will add once it's clear

ssardana08 committed:
- deleteMiner event db implemented
- bug in deleteBlobber - resolved
-

ssardana08 committed:
- unit test added for miner event db

ssardana08 committed:
- createMagicBlock updated to emit event
- Active field populated

ssardana08 committed:
- build fix

stewartie4 committed:
Merge branch 'staging' into fork_feat/miner_eventdb

ssardana96 commented:
Closing as it is from forked
